packages/eas-cli/src/project/ios/exemptEncryption.ts

```diff
@@ -0,0 +1,196 @@
+import { ExpoConfig, getConfigFilePaths } from '@expo/config';
+import { IOSConfig } from '@expo/config-plugins';
+import { getInfoPlistPathFromPbxproj } from '@expo/config-plugins/build/ios/utils/getInfoPlistPath';
+import { Platform, Workflow } from '@expo/eas-build-job';
+import plist from '@expo/plist';
+import assert from 'assert';
+import chalk from 'chalk';
+import fs from 'fs-extra';
+import path from 'node:path';
+
+import { readAppJson } from '../../build/utils/appJson';
+import Log, { learnMore } from '../../log';
+import { promptAsync } from '../../prompts';
+import { Client } from '../../vcs/vcs';
+import { getProjectConfigDescription } from '../projectUtils';
+import { resolveWorkflowAsync } from '../workflow';
+
+/** Non-exempt encryption must be set on every build in App Store Connect, we move it to before the build process to attempt only setting it once for the entire life-cycle of the project. */
+export async function ensureNonExemptEncryptionIsDefinedForManagedProjectAsync({
+  projectDir,
+  exp,
+  vcsClient,
+  nonInteractive,
+}: {
+  projectDir: string;
+  exp: ExpoConfig;
+  vcsClient: Client;
+  nonInteractive: boolean;
+}): Promise<void> {
+  const workflow = await resolveWorkflowAsync(projectDir, Platform.IOS, vcsClient);
+  assert(workflow === Workflow.MANAGED, 'This function should be called only for managed projects');
```

Comment:
we only call it for managed workflow
```ts
if (ctx.workflow === Workflow.MANAGED) {
    ...
    await ensureNonExemptEncryptionIsDefinedForManagedProjectAsync(ctx);
  }
```
so I guess we can skip it to save some execution milliseconds 🤔?